### PR TITLE
Make MCE run on AKS

### PR DESCRIPTION
**Pull request by cameronmwall**
# Description

Please provide a brief description of the purpose of this pull request.

## Related Issue

If applicable, please reference the issue(s) that this pull request addresses.

## Changes Made

Provide a clear and concise overview of the changes made in this pull request.

## Screenshots (if applicable)

Add screenshots or GIFs that demonstrate the changes visually, if relevant.

## Checklist

- [ ] I have tested the changes locally and they are functioning as expected.
- [ ] I have updated the documentation (if necessary) to reflect the changes.
- [ ] I have added/updated relevant unit tests (if applicable).
- [ ] I have ensured that my code follows the project's coding standards.
- [ ] I have checked for any potential security issues and addressed them.
- [ ] I have added necessary comments to the code, especially in complex or unclear sections.
- [ ] I have rebased my branch on top of the latest main/master branch.

## Additional Notes

Add any additional notes, context, or information that might be helpful for reviewers.

## Reviewers

Tag the appropriate reviewers who should review this pull request. To add reviewers, please add the following line: `/cc @reviewer1 @reviewer2`

## Definition of Done

- [ ] Code is reviewed.
- [ ] Code is tested.
- [ ] Documentation is updated.
- [ ] All checks and tests pass.
- [ ] Approved by at least one reviewer.
- [ ] Merged into the main/master branch.

### CODE REVIEW

File: pkg/utils/detect_ocp_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+		testEnv = &envtest.Environment{
+			CRDDirectoryPaths: []string{},
+			CRDInstallOptions: envtest.CRDInstallOptions{
+				CleanUpAfterUse: true,
+			},
+			ErrorIfCRDPathMissing: false,
+		}
+
+		cfg, err := testEnv.Start()
```
Comment (gparvin) [inline pkg/utils/detect_ocp_test.go:58]: `err` is assigned but not checked here.

File: pkg/utils/utils_test.go [hunk 2]
```diff
@@ -221,3 +221,24 @@ func TestGetHubType(t *testing.T) {
 		})
 	}
 }
+
+// func TestDetect(t *testing.T) {
```
Comment (gparvin) [inline pkg/utils/utils_test.go:225]: Are you still working on this?

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -296,3 +318,109 @@ func GetHubType(mce *backplanev1.MultiClusterEngine) string {
...
+
+func SetDeployOnOCP(v bool) {
+	GlobalDeployOnOCP = v
+}
+
+func DeployOnOCP() bool {
+	return GlobalDeployOnOCP
+}
+
+var projectGVR = schema.GroupVersionResource{Group: "project.openshift.io", Version: "v1", Resource: "projects"}
```
Comment (gparvin) [inline pkg/utils/utils.go:330]: I  bet this is left over from the previous way you were detecting OCP.

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -296,3 +317,107 @@ func GetHubType(mce *backplanev1.MultiClusterEngine) string {
...
+		if name == component {
+			return true
+		}
+	}
+	return false
+}
+
+type ServingCertGetter struct {
+	caBundleConfigMapName, namespace string
+	kubeClient                       kubernetes.Interface
```
Comment (dislbenn) [inline pkg/utils/utils.go:356]: I don't believe there is supposed to be a `,` after  `caBundleConfigMapName`.

**Pull request closed without merging**